core/modules/cart/helpers/productsEquals.ts

```diff
@@ -10,10 +15,10 @@ const getChecksum = (product: CartItem) => {
 }
 
 const getProductType = (product: CartItem): string =>
-  product.type_id || product.product_type
+  product ? product.type_id || product.product_type : null
 
 const getServerItemId = (product: CartItem): string | number =>
-  product.server_item_id || product.item_id
+  product ? product.server_item_id || product.item_id : null
 
```

Comment:
Not sure if this change is really necessary?

Reply:
maybe not, for `getProductType` it was necessary and to be certain I gave it here, although I did not encounter any issue for this function, so I will remove it:)